### PR TITLE
Filter todos on status

### PR DESCRIPTION
**Pull request by JamieZafar**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -10,12 +10,15 @@
         <header>
             <h1>TODO List</h1>
         </header>
-        <div id="count-label" class="inCompleteCounter"></div>
+        <div id="count-label" align="center" class="incomplete-counter"></div>
```
Comment (lclarksl) [inline public/index.html]: Align center should be done in CSS

File: public/main.js [hunk 2]
```diff
@@ -4,6 +4,10 @@ var form = document.getElementById("todo-form");
 var todoTitle = document.getElementById("new-todo");
 var error = document.getElementById("error");
 var countLabel = document.getElementById("count-label");
+var FILTER_ALL = 0;
+var FILTER_ACTIVE = 1;
+var FILTER_COMPLETED = 2;
+var filter = FILTER_ALL;
```
Comment (lclarksl) [inline public/main.js]: Use your set filter function here

File: public/main.css [hunk 3]
```diff
@@ -40,6 +33,22 @@ h1 {
...
+    background-color: rgba(140, 255, 83, 0.87);
+    color: #9b32ff;
+    margin-left: 5px;
+    margin-bottom: 5px;
+    border-radius: 10px;
+    box-shadow: inset 0 0 10px #000000;
+}
+
+.incomplete-counter {
+    align-content: center;
```
Comment (lclarksl) [inline public/main.css]: `align-content` should only be used with flex. Are you trying to center the content within the div or center the div itself?

Comment (JamieZafar) [inline public/main.css]: the content

File: public/main.js [hunk 4]
```diff
@@ -138,4 +162,9 @@ function completeTodo(todo) {
     createRequest.send(JSON.stringify({isComplete: !todo.isComplete}));
 }
 
+function setFilter(filterType) {
+    filter = filterType;
+    reloadTodoList();
+}
+
 reloadTodoList();
```
Comment (lclarksl) [inline public/main.js]: You don't need this `reloadTodoList` now that you're setting the filter at the top

File: public/main.js [hunk 5]
```diff
@@ -74,9 +80,27 @@ function reloadTodoList() {
     }
     todoListPlaceholder.style.display = "block";
     getTodoList(function(todos) {
+        var filteredTodos;
         todoListPlaceholder.style.display = "none";
         updateIncompletes(todos, countLabel);
-        todos.forEach(function(todo) {
+        switch (filter) {
+            case FILTER_ALL:
```
Comment (cuthullu) [inline public/main.js]: You can remove this case and just use default to catch ALL

File: public/main.css (hunk 3 above)
Comment (lclarksl) [inline public/main.css]: `text-align: center`

File: public/main.js (hunk 5 above)
Comment (lclarksl) [inline public/main.js]: Haha I told him not to do that to distinguish behaviour for invalid filters. Up to you Jamie!

Comment (cuthullu) [inline public/main.js]: YAGNI

Comment (lclarksl) [inline public/main.js]: Go with Tom

**Comment by lclarksl** [comment]
:ship:

**Comment by cuthullu** [comment]
:shipit:

**Pull request merged**